react/RatingSummary.tsx

```diff
@@ -218,7 +242,9 @@ function RatingSummary() {
         args: { authenticated: true },
       })
     })
-  }, [])
+
+    return () => eventBus.remove('reviewSaved')
+  }, [refetchTotalReviewsByProductId, refetchAverageRatingByProductId])
```

Comment:
it's not going to work. 
you must isolate that as the example below:

```ts
useEffect (() => {
   const onReviewSaved = () => {
      refetchTotalReviewsByProductId()
      refetchAverageRatingByProductId()
   }
   eventBus.on('reviewSaved', onReviewSaved);

   ...

   return () => eventBus.remove('reviewSaved', onReviewSaved)
}, [refetchTotalReviewsByProductId, refetchAverageRatingByProductId]))
```
Even though you are calling the remove by passing null as a parameter, I suggest you it to ensure that function will remove without side effects.